src/CBT.NuGet/Internal/NuGetProjectJsonParser.cs

```diff
@@ -1,53 +1,90 @@
-            {
-                if (packageRestoreData?.RestoreProjectStyle != null && !packageRestoreData.RestoreProjectStyle.Equals("ProjectJson", StringComparison.InvariantCultureIgnoreCase))
-                {
-                    yield break;
-                }
+        }
+
+        public override bool TryGetPackages(string packageConfigPath, PackageRestoreData packageRestoreData, out IEnumerable<PackageIdentityWithPath> packages)
+        {
+            packages = null;
+
+            string projectJsonPath;
 
-                // If a *proj was passed in but it is really a json project then lookup the json file.
-                if (packageRestoreData?.RestoreProjectStyle != null && packageRestoreData.RestoreProjectStyle.Equals("ProjectJson", StringComparison.OrdinalIgnoreCase))
+            if (ProjectJsonPathUtilities.IsProjectConfig(packageConfigPath))
+            {
+                projectJsonPath = packageConfigPath;
+            }
+            else
+            {
+                if (!String.Equals("ProjectJson", packageRestoreData?.RestoreProjectStyle, StringComparison.OrdinalIgnoreCase) || String.IsNullOrWhiteSpace(packageRestoreData?.ProjectJsonPath))
                 {
-                    packageConfigPath = packageRestoreData.ProjectJsonPath;
+                    return false;
                 }
-            }
 
-            if (!ProjectJsonPathUtilities.IsProjectConfig(packageConfigPath))
-            {
-                yield break;
+                projectJsonPath = packageRestoreData.ProjectJsonPath;
             }
 
-            string lockFilePath = ProjectJsonPathUtilities.GetLockFilePath(packageConfigPath);
+            string lockFilePath = ProjectJsonPathUtilities.GetLockFilePath(projectJsonPath);
 
             if (!File.Exists(lockFilePath))
             {
-                yield break;
+                throw new FileNotFoundException($"The lock file '{lockFilePath}' does not exist.  Ensure that the restore succeeded and that the lock file was generated.");
             }
 
             LockFile lockFile = LockFileUtilities.GetLockFile(lockFilePath, NullLogger.Instance);
 
-            VersionFolderPathResolver versionFolderPathResolver = new VersionFolderPathResolver(packagesPath);
+            string globalPackagesFolder = SettingsUtility.GetGlobalPackagesFolder(NuGetSettings);
 
-            foreach (LockFileLibrary library in lockFile.Libraries)
+            if (String.IsNullOrWhiteSpace(globalPackagesFolder))
             {
-                yield return new PackageIdentityWithPath(library.Name, library.Version, versionFolderPathResolver.GetPackageDirectory(library.Name, library.Version), versionFolderPathResolver.GetInstallPath(library.Name, library.Version));
+                throw new NuGetConfigurationException(@"Unable to determine the NuGet repository path.  This usually defaults to ""%UserProfile%\.nuget\packages"", ""%NUGET_PACKAGES%"", or the ""globalPackagesFolder"" in your NuGet.config.");
```

Comment:
> throw  [](start = 15, length = 7)

Be consistent warning is probably sufficient.

Reply:
I'm throwing in all tree now